### PR TITLE
Read score_precision from task.yaml and use it in the final ui

### LINKED ISSUE #119: decimal score
`task.yaml` should contain a setting for decimal digit count.

### PR DESCRIPTION
**Pull request by edomora97**
If you don't specify `score_precision` in task.yaml the UI looks pretty much identical (with the same score precision used as before). If you pass it, it will be used in some places, but not everywhere, to keep the UI compact.

Fixes #119

### CODE REVIEW

File: task-maker-format/src/ioi/finish_ui.rs [hunk 1]
```diff
@@ -389,12 +404,25 @@ impl FinishUI {
     }
 
     /// Print the score fraction of a solution using colors.
-    fn print_score_frac(&mut self, score: f64, max_score: f64) {
+    fn print_score_frac(&mut self, score: f64, max_score: f64, score_precision: Option<u8>) {
+        let score_precision = score_precision.unwrap_or(2) as usize;
```
Comment (Virv12) [inline task-maker-format/src/ioi/finish_ui.rs:408]: Why do we want two decimal places by default in the subtask's score?

**Review by Virv12 — COMMENTED** [review]
Currently we never display the score of a testcase with the correct number of digits.
I think we should use `score_precision + log10(task.max_score)` digits in the `Evaluation` sections for testcases' results, matching the number of significant digit of the final result.

File: task-maker-format/src/ioi/finish_ui.rs (hunk 1 above)
Comment (edomora97) [reply task-maker-format/src/ioi/finish_ui.rs:408]: Good point, I just replicated what we had, but your idea of score_precision + log10(max_score) makes sense.

**Comment by Virv12** [comment]
We can change `score_precision` from a `Option<u8>` to a `u8` with a default value of `0`, it looks like there's no significant difference between `None` and `Some(0)`

**Review by Virv12 — APPROVED** [review]
Looks good.

**Pull request merged**